backend/scripts/mirror_s3_data.sh

```diff
@@ -11,30 +11,80 @@ set -e
-# {dev|staging|prod} to {dev|staging} **for the public-access datasets bucket**.
-ACCOUNT_ID=`aws sts get-caller-identity | jq -r '.Arn' | grep -o '[0-9]*'`  # First get AWS account id for dev account
-SYNC_ROLE_CREDENTIALS=`aws sts assume-role --role-arn arn:aws:iam::${ACCOUNT_ID}:role/sync-datasets-dev \
-                                           --role-session-name sync-datasets \
-                       | jq -r '.Credentials'`
-# Execute sync for datasets bucket with sync role credentials
-AWS_SESSION_TOKEN=`echo $SYNC_ROLE_CREDENTIALS | jq -r '.SessionToken'` \
-AWS_SECRET_ACCESS_KEY=`echo $SYNC_ROLE_CREDENTIALS | jq -r '.SecretAccessKey'` \
-AWS_ACCESS_KEY_ID=`echo $SYNC_ROLE_CREDENTIALS | jq -r '.AccessKeyId'` \
-$S3_SYNC_CMD s3://dataset-assets-public-${SRC_ENV}/ s3://dataset-assets-public-${DEST_ENV}/
-set +x
+echo Copying WMG cube snapshot...
+
+AWS_OPTIONS="--copy-props metadata-directive --no-progress"
+S3_COPY_CMD="aws s3 cp $AWS_OPTIONS"
+S3_SYNC_CMD="aws s3 sync $AWS_OPTIONS"
+
+latest_snapshot=$(aws s3 cp s3://cellxgene-wmg-${SRC_ENV}/latest_snapshot_run -)
+if [[ $DEST_ENV == 'rdev' ]]; then
+  s3_destination="env-rdev-wmg/${STACK}"
+else
+  s3_destination="cellxgene-wmg-${DEST_ENV}"
+fi
+echo -n $latest_snapshot | aws s3 cp - s3://${s3_destination}/latest_snapshot_run
+$S3_SYNC_CMD s3://cellxgene-wmg-${SRC_ENV}/${latest_snapshot} s3://${s3_destination}/${latest_snapshot}
+
+
+echo Mirroring S3 Dataset data from $SRC_ENV to $DEST_ENV...
+
+if [[ $DEST_ENV == 'rdev' ]]; then
+  echo "Copying S3 Dataset assets for ${#COLLECTIONS[@]} Collections..."
+  DB_PW=$(aws secretsmanager get-secret-value --secret-id corpora/backend/${DEPLOYMENT_STAGE}/database --region us-west-2 | jq -r '.SecretString | match(":([^:]*)@").captures[0].string')
+
+  # Copy artifacts
+  uri_query="select uri from \"DatasetArtifact\" where id in (select unnest(artifacts) from \"DatasetVersion\" where id in (select unnest(datasets) from \"CollectionVersion\" where collection_id in ('$(sed "s/,/','/g" <<< $COLLECTIONS)')))"
+  DEPLOYMENT_STAGE=$DEPLOYMENT_STAGE make db/tunnel/up
+  uris=($(PGOPTIONS='-csearch_path=persistence_schema' PGPASSWORD=${DB_PW} psql --dbname corpora_${DEPLOYMENT_STAGE} --username corpora_${DEPLOYMENT_STAGE} --host 0.0.0.0 --csv --tuples-only -c "$uri_query"))
+  for uri in "${uris[@]}"; do
+    bucket=$(sed -E 's/s3:\/\/([^\/]+).*/\1/' <<< $uri)
+    if [[ ! -z `grep 'hosted-cellxgene' <<< $bucket` ]]; then
+      rdev_bucket_suffix="cellxgene"
+    elif [[ ! -z `grep 'corpora-data' <<< $bucket` ]]; then
+      rdev_bucket_suffix="artifacts"
+    fi
+    key=$(sed -E 's/s3:\/\/([^\/]+)\/(.*)/\2/' <<< $uri)
+    [[ "${key: -1}" == "/" ]] && s3_cmd=$S3_SYNC_CMD || s3_cmd=$S3_COPY_CMD  # use 'sync' if recursive, 'cp' otherwise
+    $s3_cmd $uri s3://env-rdev-${rdev_bucket_suffix}/${STACK}/${key} || echo "$uri does not exist -- skipping..."
+  done
+
+  # Copy public assets
```

Comment:
a bigger issue than mirroring: since rdev doesn't support cloudfront proxying to public assets, once we copy them there, can they even be accessed publicly? What is actually needed for testing purposes? If public assets can't be used testing, maybe we don't even need to support mirroring them.

if it does make sense to mirror, should it be optional?

Reply:
>since rdev doesn't support cloudfront proxying to public assets, once we copy them there, can they even be accessed publicly?

This is something @Bento007 is looking into in order to mimic prod behavior.

Reply:
>What is actually needed for testing purposes?

For Data Portal, the file must exist at that location; for rdevs with an Explorer component, then the file must be a legitimate h5ad. For these reasons, I've moved to simply using the (small) sample schema 3.0.0 Dataset for all cases. For every real asset in the prod env, an instance of the sample schema 3.0.0 Dataset will be copied to the appropriate s3 location in the rdev s3 path.